plumbum/cli/switches.py

```diff
@@ -464,18 +464,47 @@ class MyApp(Application):
                              comparison or not. The default is ``False``
     :param csv: splits the input as a comma-separated-value before validating and returning
                 a list. Accepts ``True``, ``False``, or a string for the separator
-    """
+    :param all_markers: when a user inputs any value from this set, it is considered that
+                        he requested iteration over all values. By default `all_markers`
+                        are {"*", "all"}."""
 
     def __init__(self, *values, **kwargs):
         self.case_sensitive = kwargs.pop("case_sensitive", False)
+        all_markers = kwargs.pop("all_markers", None)
+        if all_markers is None:
+            all_markers = {"*", "all"} - set(values)
```

Comment:
```suggestion
            all_markers = {"*"} - set(values)
```

Adding the English string "all" by default seems a little likely to be suprising, and possibly conflicting with potential entries. I think it would be better to default to "*" then provide an example for how to set `{"*", "all"}`.

Reply:
For "*", as you have noticed, one has to type quotes. It is not very convenient, so I usually use "all".

>Adding the English string "all" by default seems a little likely to be suprising

I don't think so: if one wants a string "all" in an arg of type "Set", it explicitly says that he wants all the values.

>, and possibly conflicting with potential entries. 

Introducing any default string can be conflicting. That's why the line to which you have left this comment exists - to avoid the conflicts automatically. If user-provided Set doesn't have `all` and `*`, then theu will be used as markers for all the values.

Here is the only problem, I have hardcoded them, it is not good.